src/app/App.js

```diff
@@ -284,8 +284,13 @@ class App extends EventEmitter {
 	 * @return {boolean}
 	 */
 	canNavigate(url) {
-		var path = utils.getUrlPath(url);
-		var uri = new Uri(url);
+		const uri = utils.validateUrl(url);
```

Comment:
I honestly don't like that at all... it makes reading the code very hard because the object has different shapes, so you actually need to dive into the method implementation to see what's being returned.